src/compiler/checker.ts

```diff
@@ -2904,16 +2904,17 @@ module ts {
         }
 
         function getPropertiesOfType(type: Type): Symbol[] {
-            if (type.flags & TypeFlags.Union) {
-                return getPropertiesOfUnionType(<UnionType>type);
-            }
-            return getPropertiesOfObjectType(getApparentType(type));
+            type = getApparentType(type);
+            return type.flags & TypeFlags.Union ? getPropertiesOfUnionType(<UnionType>type) : getPropertiesOfObjectType(type);

```

Comment:
The problem with long single lines is github:

![image](https://cloud.githubusercontent.com/assets/4564579/7123448/f156d1e2-e1d6-11e4-958a-acbf718489bd.png)

It's actually harder to read because github is optimized for 80 columns.